src/SB/Core/gc/iModel.cpp

```diff
@@ -49,15 +76,65 @@ void iModelUnload(RpAtomic* userdata)
     if ((RpClump*)userdata != 0)
     {
         RwFrameGetRoot((RwFrame*)userdata);
+
+        userdata = 0;
+    }
+    else
+    {
+        RwFrameDestroyHierarchy(frame);
+
         RwFrameDestroyHierarchy(frame);
         userdata = 0;
+
     }
     if (clump != 0)
     {
         RpClumpDestroy(clump);
     }
 }
 
+
+RpAtomic* NextAtomicCallback(RpAtomic* atomic, void* data)
+{
+    // What?
+
+    RpAtomic** nextModel;
+
+    if (data != atomic)
+    {
+        atomic = 0;
+        return atomic;
+    }
+    if (data != 0)
+    {
+        return atomic;
+    }
+
+    atomic = (RpAtomic*)data;
+
+    return atomic;
+}
+
+RpAtomic* iModelFile_RWMultiAtomic(RpAtomic* model)
+{
+    RpClump* clump;
+    RpAtomic* nextModel = 0;
+    RpAtomic* modelArray[2];
+
+    if (model == 0)
+    {
+        modelArray[0] = 0;
+    }
+    else
+    {
+        //modelArray[0] = (S32)model;
```

Comment:
Please remember to take out commented code - it adds unnecessary noise when trying to debug functions to increase the match.